presto-main/src/test/java/io/prestosql/operator/scalar/TestDateTimeFunctionsBase.java

```diff
@@ -1131,6 +1133,31 @@ public void testIntervalDayToSecondToMilliseconds()
         assertFunction("to_milliseconds(parse_duration('1d'))", BigintType.BIGINT, DAYS.toMillis(1));
     }
 
+    @Test
+    public void testWithTimezone()
+    {
+        String timestamp = "2001-08-22 03:04:05.321";
+        assertAtTimezone(timestamp, "UTC");
+        assertAtTimezone(timestamp, "+13");
+        assertAtTimezone(timestamp, "-14");
+        assertAtTimezone(timestamp, "+00:45");
+        assertAtTimezone(timestamp, "Asia/Shanghai");
+        assertAtTimezone(timestamp, "America/New_York");
+
+
+        assertAtTimezone("2001-06-01 03:04:05.321", "America/Los_Angeles");
+        assertAtTimezone("2001-12-01 03:04:05.321", "America/Los_Angeles");
+
+        assertInvalidFunction("with_timezone(TIMESTAMP '2001-08-22 03:04:05.321', 'invalidzoneid')", NOT_SUPPORTED);
+    }
+
+    private void assertAtTimezone(String timestampString, String zoneId)
+    {
+        String timestamp = format("TIMESTAMP '%s'", timestampString);
+        String timestampWithTimezone = format("TIMESTAMP '%s %s'", timestampString, zoneId);
+        assertFunction(format("with_timezone(%s, '%s') = %s", timestamp, zoneId, timestampWithTimezone), BooleanType.BOOLEAN, true);
```

Comment:
When this fails, this will only print "Expected true, got false".
Please use `assertFunction(String projection, Type expectedType, Object expected)` overload.

(See `testTimeWithTimeZoneAtTimeZone` for an example how you can construct expected `SqlTimeWithTimeZone`.)